lib/Jit/jitoptions.cpp

```diff
@@ -50,11 +58,52 @@ void freeStringConfigValue(ICorJitInfo *CorInfo, UTF16CharT *Value) {
 }
 
 JitOptions::JitOptions(LLILCJitContext &Context) {
-  // Set 'IsAltJit' based on environment information.
-  IsAltJit = queryIsAltJit(Context);
+
+  if (!StaticOptionsInitialized) {
+    StaticDumpLevel = ::DumpLevel::NODUMP;
+
+    char16_t *LevelWStr =
+      getStringConfigValue(Context.JitInfo, UTF16("DUMPLLVMIR"));
+    if (LevelWStr) {
+      std::unique_ptr<std::string> Level = Convert::utf16ToUtf8(LevelWStr);
+      std::transform(Level->begin(), Level->end(), Level->begin(), ::toupper);
+      if (Level->compare("VERBOSE") == 0) {
+        StaticDumpLevel = ::DumpLevel::VERBOSE;
+
+      } else if (Level->compare("SUMMARY") == 0) {
+        StaticDumpLevel = ::DumpLevel::SUMMARY;
+      }
+    }
+    if (Context.Flags & CORJIT_FLG_PREJIT) {
+      InitMethodSet(Context, AltJitNgenMethodSet, UTF16("AltJitNgen"));
+      // Set up AltJitNgen set to be used for AltJit test.
+      AltJitPtr = &AltJitNgenMethodSet;
+    } else {
+      InitMethodSet(Context, AltJitMethodSet, UTF16("AltJit"));
+      // Set up AltJit set to be use for AltJit test.
+      AltJitPtr = &AltJitMethodSet;
+    }
+    InitMethodSet(Context, ExcludeMethodSet, UTF16("AltJitExclude"));
+    InitMethodSet(Context, BreakMethodSet, UTF16("AltJitBreak"));
+    InitMethodSet(Context, MSILMethodSet, UTF16("AltJitMSILDump"));
+    InitMethodSet(Context, LLVMMethodSet, UTF16("AltJitLLVMDump"));
+    InitMethodSet(Context, EntrySizeMethodSet, UTF16("AltJitCodeRangeDump"));
+    StaticOptionsInitialized = true;
+  }
+
 
   // Set dump level for this JIT invocation.
-  DumpLevel = queryDumpLevel(Context);
+  DumpLevel = StaticDumpLevel;
+  const char *ClassName = nullptr;
+  const char *MethodName = 
+    Context.JitInfo->getMethodName(Context.MethodInfo->ftn, &ClassName);
+  PCCOR_SIGNATURE PCSig = Context.MethodInfo->args.pSig;
+  IsAltJit = AltJitPtr->contains(MethodName, ClassName, PCSig);
+  IsExcludeMethod = ExcludeMethodSet.contains(MethodName, ClassName, PCSig);
+  IsBreakMethod = BreakMethodSet.contains(MethodName, ClassName, PCSig);
+  IsMSILDumpMethod = MSILMethodSet.contains(MethodName, ClassName, PCSig);
+  IsLLVMDumpMethod = LLVMMethodSet.contains(MethodName, ClassName, PCSig);
+  IsEntrySizeMethod = EntrySizeMethodSet.contains(MethodName, ClassName, PCSig);
```

Comment:
The explanation for the change is that it is more efficient to check for the once-only 
initialization in just one place, rather than in multiple places. 

In addition, the previous methods of checking for initialization of MethodSets was to
use their "isInitialized" method, but for MethodSets that are not specified these
would never be initialized. In any case I think it is more confusing to have lots of
little "isInitialized" flags than to have just one, since all the options are
initialized at once (the first time LLILC is invoked).

The reason to change the code for AltJit is that it is a MethodSet so we should
handle it uniformly in the same way we handle the other MethodSets.

The refactoring is also more efficient each time JitOptions is constructed since
one call to getMethodName serves for all the calls to MethodSet.contains.

I didn't do a completely thorough job of converting all of the options to this approach.
That is why there still are some remaining separate query methods that remain.
I was going to complete the job in a subsequent check-in, though that is
not essential.